### PR TITLE
use Eloquent::when() – no more if-else’s

### PR DESCRIPTION
**Pull request by RoisNewVersion**
use Eloquent::when() method instead of using if else statement.

### CODE REVIEW

**Comment by nafiesl** [comment]
@RoisNewVersion, thanks for the PR. Could you please fix your code to pass the Style-CI? I see you change a lot of string concatenations.

**Comment by RoisNewVersion** [comment]
Sorry for interrupt, I used to have php formater ext in my VSCode, I just hit Shift+Alt+F that automatically reformat the code.
as I see, the styleci didn't accept white space between string concatenation, am I right? So, what should I do then?

**Comment by nafiesl** [comment]
@RoisNewVersion, you can turn off your auto format from your VSCode, then just change the lines that you want to change.

If we passed the style-ci, I will start reviewing your changes.

**Comment by RoisNewVersion** [comment]
Okay, I did it.

File: app/Http/Controllers/Controller.php [hunk 1]
```diff
@@ -112,11 +115,13 @@ protected function getCategoryTransactions(Category $category, array $criteria)
         $partnerId = $criteria['partner_id'];
 
         $transactionQuery = $category->transactions();
-        $transactionQuery->where('description', 'like', '%'.$query.'%');
         $transactionQuery->whereBetween('date', [$startDate, $endDate]);
-        if ($partnerId) {
-            $transactionQuery->where('partner_id', $partnerId);
-        }
+        $transactionQuery->when($query, function ($q, $query) {
```
Comment (nafiesl) [inline app/Http/Controllers/Controller.php]:
```suggestion
        $transactionQuery->when($query, function ($queryBuilder, $query) {
```

File: app/Http/Controllers/Controller.php [hunk 2]
```diff
@@ -112,11 +115,13 @@ protected function getCategoryTransactions(Category $category, array $criteria)
...
+            $q->where('description', 'like', '%'.$query.'%');
```
Comment (nafiesl) [inline app/Http/Controllers/Controller.php]:
```suggestion
            $queryBuilder->where('description', 'like', '%'.$query.'%');
```

File: app/Http/Controllers/Controller.php [hunk 3]
```diff
@@ -112,11 +115,13 @@ protected function getCategoryTransactions(Category $category, array $criteria)
...
+        });
+        $transactionQuery->when($partnerId, function ($w, $partnerId) {
```
Comment (nafiesl) [inline app/Http/Controllers/Controller.php]:
```suggestion
        $transactionQuery->when($partnerId, function ($queryBuilder, $partnerId) {
```

File: app/Http/Controllers/Controller.php [hunk 4]
```diff
@@ -112,11 +115,13 @@ protected function getCategoryTransactions(Category $category, array $criteria)
...
+            $w->where('partner_id', $partnerId);
```
Comment (nafiesl) [inline app/Http/Controllers/Controller.php]:
```suggestion
            $queryBuilder->where('partner_id', $partnerId);
```

File: app/Http/Controllers/Controller.php [hunk 5]
```diff
@@ -136,11 +141,13 @@ protected function getPartnerTransactions(Partner $partner, array $criteria)
         $categoryId = $criteria['category_id'];
 
         $transactionQuery = $partner->transactions();
-        $transactionQuery->where('description', 'like', '%'.$query.'%');
-        $transactionQuery->whereBetween('date', [$startDate, $endDate]);
```
Comment (nafiesl) [inline app/Http/Controllers/Controller.php]: Why are you removing this line, @RoisNewVersion? This line is still needed.

File: app/Http/Controllers/Controller.php [hunk 6]
```diff
@@ -136,11 +141,13 @@ protected function getPartnerTransactions(Partner $partner, array $criteria)
...
-        if ($categoryId) {
-            $transactionQuery->where('category_id', $categoryId);
-        }
+
+        $transactionQuery->when($query, function ($q, $query) {
```
Comment (nafiesl) [inline app/Http/Controllers/Controller.php]:
```suggestion
        $transactionQuery->when($query, function ($queryBuilder, $query) {
```

File: app/Http/Controllers/Controller.php [hunk 7]
```diff
@@ -136,11 +141,13 @@ protected function getPartnerTransactions(Partner $partner, array $criteria)
...
+            $q->where('description', 'like', '%'.$query.'%');
```
Comment (nafiesl) [inline app/Http/Controllers/Controller.php]:
```suggestion
            $queryBuilder->where('description', 'like', '%'.$query.'%');
```

File: app/Http/Controllers/Controller.php [hunk 8]
```diff
@@ -136,11 +141,13 @@ protected function getPartnerTransactions(Partner $partner, array $criteria)
...
+        });
+        $transactionQuery->when($categoryId, function ($w, $categoryId) {
```
Comment (nafiesl) [inline app/Http/Controllers/Controller.php]:
```suggestion
        $transactionQuery->when($categoryId, function ($queryBuilder, $categoryId) {
```

File: app/Http/Controllers/Controller.php [hunk 9]
```diff
@@ -136,11 +141,13 @@ protected function getPartnerTransactions(Partner $partner, array $criteria)
...
+            $w->where('category_id', $categoryId);
```
Comment (nafiesl) [inline app/Http/Controllers/Controller.php]:
```suggestion
            $queryBuilder->where('category_id', $categoryId);
```

File: app/Http/Controllers/Controller.php (hunk 5 above)
Comment (RoisNewVersion) [reply app/Http/Controllers/Controller.php]: Oh man, Sorry, That was my mistake. I always running phpunit test when done change something, and it shows me green.
Thanks for your corrections.

Comment (RoisNewVersion) [reply app/Http/Controllers/Controller.php]: Sorry for late update, I was busy. May you review the code. Thanks.

**Comment by nafiesl** [comment]
@RoisNewVersion, why are you closing this PR?

**Pull request merged**